src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -271,18 +271,18 @@ export class ExtHostTerminal extends Disposable {
 		let extHostType: vscode.TerminalShellType | undefined;
 
 		switch (shellType) {
-			case 'sh': extHostType = 1; break;
-			case 'bash': extHostType = 2; break;
-			case 'fish': extHostType = 3; break;
-			case 'csh': extHostType = 4; break;
-			case 'ksh': extHostType = 5; break;
-			case 'zsh': extHostType = 6; break;
-			case 'cmd': extHostType = 7; break;
-			case 'gitbash': extHostType = 8; break;
-			case 'pwsh': extHostType = 9; break;
-			case 'python': extHostType = 10; break;
-			case 'julia': extHostType = 11; break;
-			case 'nu': extHostType = 12; break;
+			case PosixShellType.Sh: extHostType = 1; break;
+			case PosixShellType.Bash: extHostType = 2; break;
+			case PosixShellType.Fish: extHostType = 3; break;
+			case PosixShellType.Csh: extHostType = 4; break;
```

Comment:
Looking at this now, you'd be better off just copying the enum from the API and duplicating it here. You could add a comment on top stating that it's important it keeps in sync

Reply:
Did you mean something like [this](https://github.com/microsoft/vscode/pull/237624/commits/9827ad1b817d5b5deac93adb705a506703e7c943)? 

Or having something like `case vscode.TerminalShellType.Sh: extHostType = 1; break;`?

Reply:
Like this:

```ts
                // IMPORTANT: This must stay in sync with the extension host API type.
		enum VscodeTerminalShellType {
			Sh = 1,
			Bash,
			Fish,
			Csh,
			Ksh,
			Zsh,
			CommandPrompt,
			GitBash,
			PowerShell,
			Python,
			Julia,
			NuShell
		}
		switch (shellType) {
			case PosixShellType.Sh: extHostType = VscodeTerminalShellType.Sh; break;
			case PosixShellType.Bash: extHostType = VscodeTerminalShellType.Bash; break;
```

Reply:
We're duplicating because we're not allowed to import anything but types from `vscode`. An enum isn't just a type but it also has an implementation (sh = 1, etc.).